src/Entities/Entity.cpp

```diff
@@ -1260,9 +1261,25 @@ void cEntity::DetectCacti(void)
 
 
 
+void cEntity::ScheduleMoveToWorld(cWorld *a_World, Vector3d a_NewPosition)
+{
+	m_NewWorld = a_World;
+	m_NewWorldPosition = a_NewPosition;
+	m_ReadyToChangeWorlds = true;
+}
+
+
+
 
 bool cEntity::DetectPortal()
 {
+	if (m_ReadyToChangeWorlds)
+	{
+		m_ReadyToChangeWorlds = false;
+		MoveToWorld(m_NewWorld, false, m_NewWorldPosition);
+		return true;
+	}
```

Comment:
I think a simple `/* If a move to another world was scheduled via ScheduleMoveToWorld(), move the entity now. */` would have been enough. This is too wordy and still doesn't explain much. Also note that comments inside functions shouldn't use the doxy-comment format (`/**`), only comments above declarations should use it.

Also the `m_ReadyToChangeWorlds` is a somewhat misleading name, `m_IsWorldChangeScheduled` might have been better.